src/MissionManager/MissionController.cc

```diff
@@ -61,8 +61,8 @@ MissionController::MissionController(PlanMasterController* masterController, QOb
     connect(this,                                               &MissionController::missionPlannedDistanceChanged,      this, &MissionController::recalcTerrainProfile);
 
     // The follow is used to compress multiple recalc calls in a row to into a single call.
-    connect(this, &MissionController::_recalcMissionFlightStatusSignal, this, &MissionController::_recalcMissionFlightStatus,   Qt::QueuedConnection);
-    connect(this, &MissionController::_recalcFlightPathSegmentsSignal,  this, &MissionController::_recalcFlightPathSegments,    Qt::QueuedConnection);
+    connect(this, &MissionController::_recalcMissionFlightStatusSignal, this, &MissionController::_recalcMissionFlightStatus);
```

Comment:
This might fix the symptom but it will break other things. The compressing of multiple signals into a signals is there for performance reasons.

Reply:
I did a test on how often `_recalcMissionFlightStatus` gets called every time I click `Download` to re-download the plan from the vehicle.

When connect uses `Qt::QueuedConnection`: 5 calls per click
When connect does NOT use  `Qt::QueuedConnection`: 27 calls per click

So there is indeed some resource saving when using the `Qt::QueuedConnection` flag.